### PR TITLE
RP-694: warn on duplicate tenant IDs

### PR DESCRIPTION
**Pull request by gregcharles**

### CODE REVIEW

File: webapp/src/main/webapp/src/app/admin/tenant/pages/tenant/tenant.component.ts [hunk 1]
```diff
@@ -83,6 +84,11 @@ export class TenantComponent implements OnDestroy {
         .exists((value || '').toUpperCase())
         .pipe(map(exists => !exists));
 
+    this.tenantIdAvailable = (value: string) =>
+      this.service
+        .existsId((value || '').toUpperCase())
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/tenant/pages/tenant/tenant.component.ts]: @ftdholbrook do we force uppercase on IDs too? i cant remember. if we allow mixed case i think we would do an exact case match

Comment (ftdholbrook) [reply webapp/src/main/webapp/src/app/admin/tenant/pages/tenant/tenant.component.ts]: I don't think we are forcing anything on ID's but I can double check.

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/administration/TenantAdministrationStatusService.java [hunk 2]
```diff
@@ -67,6 +67,14 @@
      */
     Optional<TenantAdministrationStatusContext> findStatus(String tenantKey);
 
+    /**
+     * Checks if an tenant with the given ID already exists.
+     *
+     * @param id the tenant ID
+     * @return true if the tenant already exists, false if not
+     */
+    boolean checkId(String id);
```
Comment (malaffoon) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/administration/TenantAdministrationStatusService.java]: This seems out of place? Perhaps i'll see further on but this has nothing to do with status does it?

Comment (malaffoon) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/administration/TenantAdministrationStatusService.java]: Hmm, as part of the status service should this be more like `Optional<...> findStatusById(String tenantId)`?

**Comment by malaffoon** [comment]
Does it bother anybody else that it requires enhancing three services to do this?

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/administration/TenantAdministrationStatusService.java (hunk 2 above)
Comment (gregcharles) [reply admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/administration/TenantAdministrationStatusService.java]: I'd be more inclined to change the other method. The status is never actually used, except to determine its presence.

**Comment by gregcharles** [comment]
> Does it bother anybody else that it requires enhancing three services to do this?

Ha, it bothered me! :) Less now that the work is done though.

**Comment by gregcharles** [comment]
Updated status service to return status by ID instead of a boolean.

**Pull request merged**